src/codeu/chat/client/commandline/Chat.java

```diff
@@ -403,18 +407,25 @@ public void invoke(List<String> args) {
     panel.register("m-list", new Panel.Command() {
       @Override
       public void invoke(List<String> args) {
-        System.out.println("--- start of conversation ---");
-        for (MessageContext message = conversation.firstMessage();
-                            message != null;
-                            message = message.next()) {
-          System.out.println();
-          System.out.format("USER : %s\n", message.message.author);
-          System.out.format("SENT : %s\n", message.message.creation);
-          System.out.println();
-          System.out.println(message.message.content);
-          System.out.println();
+        if(conversation.conversation.getAccessLevel(conversation.user).hasMemberAccess()
+                || conversation.conversation.getAccessLevel(conversation.user).hasOwnerAccess()
+                || conversation.conversation.getAccessLevel(conversation.user).hasCreatorAccess()){
```

Comment:
Multiple if's not needed because AL is written to note that AccessLevels are subsets of each other. Creator has creator, owner and member access. Owner has owner and member access.